backend/Api/Consultants/ConsultantExtensions.cs

```diff
@@ -26,10 +27,76 @@ public static ConsultantReadModel MapConsultantToReadModel(this Consultant consu
             yearsOfExperience,
             consultant.Degree ?? Degree.None,
             bookedHours,
+            detailedBooking,
             isOccupied
         );
     }
 
+    private static List<DetailedBooking> GetDetailedBooking(Consultant consultant, Week firstWeek, int weeks)
+    {
+        var org = consultant.Department.Organization;
+
+        var hoursPrWorkDay = org.HoursPerWorkday;
+
+        var weeksInSet = DateService.GetNextWeeks(firstWeek, weeks);
+        
+        var projects = consultant.Staffings
+            .Where(s => weeksInSet.Contains(new Week(s.Year, s.Week)))
+            .Where(s=> s.Project.State == ProjectState.Active || s.Project.State == ProjectState.Offer)
+            .GroupBy(s => s.Project.Customer.Name)
+            .Select(grouping => new DetailedBooking(
```

Comment:
Jeg ser det er en del likheter mellom denne nøstet Select og den på linje 62 for plannedAbsences. Kan logikken her trekkes ut i en hjelpefunksjon for å redusere duplikat kode?

Reply:
Det kan det nok! Planen er egentlig en endring av hele greia her, fordi det er sååååå tregt. Jeg har lekt litt, og tatt ned kjøretiden fra ca 6-10 sekund til 30-80 millisekund. Så vi skriver om hele greia og da blir ting forhåpentligvis penere og